### PR TITLE
support background color for SectionNumber widget

### PR DESCRIPTION
**Pull request by revahh**
## Related Issues
Fixes <https://github.com/demisto/etc/issues/29416>

## Related PRs
branch | PR
------ | ------
server PR | <https://github.com/demisto/server/pull/16634>

add background color in case of widget color threshold is enabled

### CODE REVIEW

File: src/components/Sections/SectionNumber.less [hunk 1]
```diff
@@ -21,7 +23,9 @@
         padding: 3px 10px 3px 5px;
         background-color: #FFF;
         color: #555555;
-        box-shadow: inset 0 1px 0 0 rgba(255, 255, 255, 0.13), 0 4px 5px 0 rgba(0, 0, 0, 0.1);
+        &.no-custom-background {
+          box-shadow: inset 0 1px 0 0 rgba(255, 255, 255, 0.13), 0 4px 5px 0 rgba(0, 0, 0, 0.1);
```
Comment (Bars92) [inline src/components/Sections/SectionNumber.less:27]: i think we can remove this part

File: src/components/Sections/SectionNumber.js [hunk 2]
```diff
@@ -64,7 +64,7 @@ const SectionNumber = ({ data, layout, style, sign, signAlignment, title, titleS
         {layout === CHART_LAYOUT_TYPE.horizontal && isTrend &&
         trendContainer
         }
-        <div className="trend-message" style={titleStyle}>
+        <div className="trend-message" style={{ color: titleColor }}>
```
Comment (Bars92) [inline src/components/Sections/SectionNumber.js:67]:
```suggestion
        <div className="trend-message" style={{ ...titleStyle, color: titleColor }}>
```

File: src/components/Sections/SectionNumber.js [hunk 3]
```diff
@@ -31,13 +31,15 @@ const SectionNumber = ({ data, layout, style, sign, signAlignment, title, titleS
     shortPercentage = `>${TREND_NUMBER_LIMIT}`;
   }
 
+  const color = style && style.backgroundColor ? '#FFF' : undefined;
+  const titleColor = (style.backgroundColor || (titleStyle && titleStyle.color)) ? '#FFF' : undefined;
```
Comment (Bars92) [inline src/components/Sections/SectionNumber.js:35]: can't style be undefined ?

File: src/components/Layouts/ReportLayout.less [hunk 4]
```diff
@@ -63,7 +63,7 @@ h1 {
       overflow: visible;
       transition: none;
       z-index: 1;
-      height: auto !important;
+      height: auto;
```
Comment (Bars92) [inline src/components/Layouts/ReportLayout.less:66]: i think this is used for the layout fixes. have you checked out all test cases?

Comment (revahh) [reply src/components/Layouts/ReportLayout.less:66]: tested several cases, can you point me to specific test cases?

Comment (Bars92) [reply src/components/Layouts/ReportLayout.less:66]: lets revert this change for now

File: src/components/Sections/SectionNumber.js [hunk 5]
```diff
@@ -31,6 +31,8 @@ const SectionNumber = ({ data, layout, style, sign, signAlignment, title, titleS
     shortPercentage = `>${TREND_NUMBER_LIMIT}`;
   }
 
+  const color = style && style.backgroundColor ? '#FFF' : undefined;
+  const titleColor = (color || (titleStyle && titleStyle.color)) ? '#FFF' : undefined;
```
Comment (Bars92) [inline src/components/Sections/SectionNumber.js:35]: we still want to enable `titleStyle.color` so maybe
```
get(titleStyle, 'color') || color
```

**Review by Bars92 — COMMENTED** [review]
mind small comments so we can keep supporting titleStyle.color

**Review by Bars92 — APPROVED** [review]

**Pull request merged**